### ISSUE TITLE
[Feature] enable modular development like springboot modulith or google service weaver

### ISSUE THREAD

**Issue by lvjing2** [apache/dubbo#14598]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dubbo/issues?q=is%3Aissue) and found no similar feature requirement.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Descriptions

Currently, there are many concerns in modular development, no matter springboot modulith or google service weaver. We notice that we can achieve those benefit by the modular development.
1. developing with only focus on his own business
2. deploy much faster
3. cost less resource
4. more simper cooperation with other developers
5. and so on
 
So, How to enable modular development in dubbo? Can we discompose the mess  code into small and separated modulars，develop and deploy separately (hot deploy).

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by oxsean** [comment]
Dubbo supports multi-layer isolation with ScopeModel https://cn.dubbo.apache.org/zh-cn/blog/1/01/01/dubbo-3-%E4%B8%AD%E7%9A%84%E4%B8%89%E5%B1%82%E9%85%8D%E7%BD%AE%E9%9A%94%E7%A6%BB/ and already provides dynamic service export and unexport. For hot deployment, you only need to properly handle dubbo service provider destroy and recreation during module updates..

**Comment by lvjing2** [comment]
Yes, ScopeModel provided the isolation in runtime, but for modular development it's maybe need not only isolation in runtime, like
1. support the code of different modulars in different repo

And also there are still many tools needed to get fully benefit of modular development, which from our practical. like
1. module package into seperated module jar
2. operation tunnel to install and uninstall the module
3. communication between models
4. modular serving system like k8s
5. how to migrate multiple old dubbo apps into one jvm or one base dubbo app
and so on

which we can find in [koupleless.io](https://koupleless.io/docs/tutorials/module-development/module-slimming/)
